### PR TITLE
KAA-934 CMocka framework integration

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

**Comment by forGGe** [comment]
Do not review! Merge conflicts must be addressed

File: client/client-multi/client-c/test/platform-impl/test_ext_log_storage_memory.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+/**
```
Comment (rasendubi) [inline client/client-multi/client-c/test/platform-impl/test_ext_log_storage_memory.c]: That's not a doxygen comment

**Comment by rasendubi** [comment]
Please, rebase on the current master. Newly added test (`test_get_request_kaasync_over_buff`) should be adapted to cmocka.

**Comment by forGGe** [comment]
Rebased, fixed test.
/cc @rasendubi @dyosick

File: client/client-multi/client-c/test/test_kaa_common.c [hunk 2]
```diff
@@ -26,8 +26,10 @@
 
 static kaa_logger_t *logger = NULL;
 
-void test_profile_update(void)
+void test_profile_update(void **state)
 {
+    (void)state;
+    (void) state;
```
Comment (ghost) [inline client/client-multi/client-c/test/test_kaa_common.c]: Duplicates.

**Comment by forGGe** [comment]
Review addressed

**Comment by rasendubi** [comment]
LGTM

**Pull request merged**